src/screens/payment-methods/PaymentMethods.tsx

```diff
@@ -150,15 +155,84 @@ export default class PaymentMethods extends BaseAutoRefreshScreen<Props, State>
               count={count}
               limit={limit}
               skip={skip}
-              renderItem={(item: BillingPaymentMethod) => <PaymentMethodComponent paymentMethod={item} navigation={navigation} />}
+              renderItem={(paymentMethod: BillingPaymentMethod) => (
+                <Swipeable
+                  overshootRight={false}
+                  overshootLeft={false}
+                  childrenContainerStyle={style.swiperChildrenContainer}
+                  renderRightActions={() => this.renderSwipeableActions(paymentMethod, style)}>
+                  <PaymentMethodComponent paymentMethod={paymentMethod} navigation={navigation} />
+                </Swipeable>
+              )}
               refreshing={refreshing}
               manualRefresh={this.manualRefresh}
               onEndReached={this.onEndScroll}
-              emptyTitle={i18n.t('paymentMethods.noPaymentMethod')}
+              emptyTitle={I18n.t('paymentMethods.noPaymentMethod')}
             />
           </View>
         )}
       </Container>
     );
   };
+
+  private renderSwipeableActions(paymentMethod: BillingPaymentMethod, style: any) {
+    const deleteInProgress = this.state.deleteOperationsStates?.[paymentMethod.id];
+    const commonColors = Utils.getCurrentCommonColor();
+    return paymentMethod.isDefault ? null : (
+      <TouchableOpacity
+        disabled={deleteInProgress}
+        style={style.trashIconButton}
+        onPress={() => {
+          this.onPressDelete(paymentMethod);
```

Comment:
To rename 'onPressDelete' to 'deletePaymentMethodConfirm' to be aligned with methods in ChargingStationConnectorDetails.tsx ?